tests/test_metrics.py

```diff
@@ -0,0 +1,55 @@
+import mock
+import pytest
+
+from gradient_utils.metrics import get_metric_pushgateway, CollectorRegistry, add_metrics
+
+@mock.patch("gradient_utils.metrics.push_to_gateway")
+@mock.patch("gradient_utils.metrics.Gauge")
+def test_add_metrics_pushes_metrics(gauge_patched, push_to_gateway_patched):
+    registry = CollectorRegistry()
+    metrics = {
+        'cat': 1,
+        'dog': 2,
+        'catdog': 1.2
+    }
+    add_metrics(metrics, workload_id='some_id', registry=registry)
+
+    # TODO: The best way to test would be to verify that the three metrics are
+    #       waiting at the pushgateway.
```

Comment:
If you want to do integration testing in CI, you'll need to run a second container next to our test job in `.circleci/config.yml`. The push gateway we use is already containerized.

Reply:
So I used the official prom pushgateway since it has admin access to get and clear the gateway where the weaveworks one doesnt. 
Since they have the same interface, I think it's okay for this library since we're not concerned with what the push gateway does under the hood, just that it received it. WDYT?

Reply:
Ideally our integration test would map to the real-world case. I'm not sure why we went with the weaveworks implementation in our clusters. Maybe @paperspace-philip knows more.